test/types/schema.test.ts

```diff
@@ -377,9 +377,9 @@ export function autoTypedSchema() {
     mixed1?: any;
     mixed2?: any;
     mixed3?: any;
-    objectId1?: Schema.Types.ObjectId;
-    objectId2?: Schema.Types.ObjectId;
-    objectId3?: Schema.Types.ObjectId;
+    objectId1?: Types.ObjectId;
+    objectId2?: Types.ObjectId;
```

Comment:
This is wrong.

Reply:
Okay... Just these ones or the `Decimal128` as well?

Or do you mean like every where I changed from `Schema.Types` to `Types`?

Reply:
basically everywhere where you did it.

Reply:
Damn... What about the TS errors about the `Schema.Types.ObjectId` not matching the `Types.ObjectId` though?

Like in #12002, I had the errors as well.

Reply:
hmm. now you make me wonder if this is actually correct.

@mohammad0-0ahmad 

Can you give your opinion on this?

Reply:
Let me check that

Reply:
According to https://mongoosejs.com/docs/typescript.html#objectids-and-other-mongoose-types the changes made in test file are correct, but I don't know if that the same in inferschematype.
I need more time.
I would like to say that this PR still have TS error that appear by running "npm run test-tsd".
I will come back with more info.
@iammola  thanks for PR and try to fix the mentiond error please.

Reply:
I honestly have no idea where the error is still from. I've just tested the types on my system and they work fine.

![image](https://user-images.githubusercontent.com/55806855/176561529-5349fe7e-1dfd-4646-ad18-154013628807.jpeg)

Can you please help me check it on your end?

The only thing I'll change is the type of the mixed properties to `unknown` from `any`. But that doesn't seem like it'll make a difference

Reply:
To be honest, I don't think It is a good idea to make it unknown instead of any https://mongoosejs.com/docs/schematypes.html#mixed
I need to understand the issue more, so just give more time :)

Reply:
Alright... I only used `unknown` because it was the last type/else value if the path didn't match any other types.

https://github.com/Automattic/mongoose/blob/13d26c360c8a91bf8c537399bb9feb1713be7d82/types/inferschematype.d.ts#L153-L155